### ISSUE TITLE
Prototype profiler object serialization using JSON

### ISSUE THREAD

**Issue by yungyuc** [solvcon/solvcon#343]
This issue derives from #302.

Develop a helper class or some helper classes to serialize [the `CallProfile` object](https://github.com/solvcon/modmesh/blob/69b7e899bcb554cf6068951f6c79048017e2b7fe/cpp/modmesh/toggle/RadixTree.hpp#L178) used for recording the profile data (in memory).  Remarks:

1. The helper class(es) should use proper encapsulation.
2. The serialized data should be testable in Python.  It may also be tested in C++ (google-test) but the C++ tests are not mandatory.
3. The serialized JSON string should be human readable.  The readability should be easy and use decent indentation.

There should be a capability to deserialize the data back to `CallProfile`.  We may use this issue to track the deserialization, or use a different issue.

---

- [ ] Make more member data for serialization.  (See https://github.com/solvcon/modmesh/issues/343#issuecomment-2308388291)
- [ ] Implement deserializer.  (See https://github.com/solvcon/modmesh/issues/343#issuecomment-2308388291)

**Comment by yungyuc** [comment]
@ThreeMonth03 is working on this issue with PR #323.

**Comment by yungyuc** [comment]
@ThreeMonth03 plans to make the next review for PR #323 ready by 5th June.

**Comment by yungyuc** [comment]
#323 is merged.  We should start to exercise the data for analyzing performance to find out how useful it is and how to improve the design (of the profiler and the serialization).

**Comment by tigercosmos** [comment]
@ThreeMonth03 Since https://github.com/solvcon/modmesh/pull/372 and https://github.com/solvcon/modmesh/pull/377, the general JSON parser, is merged, could you revise `CallProfiler` to use the new `SerializableItem`. For more details and examples, you can check the https://github.com/solvcon/modmesh/pull/372 and https://github.com/solvcon/modmesh/pull/377.

**Comment by ThreeMonth03** [comment]
> @ThreeMonth03 Since #372 and #377, the general JSON parser, is merged, could you revise `CallProfiler` to use the new `SerializableItem`. For more details and examples, you can check the #372 and #377.

Thank's for your contribution, I will trace the implementation and revise the `CallProfiler`.

**Comment by yungyuc** [comment]
@ThreeMonth03 @q40603 @tigercosmos do we have anything needed for this issue or need any further PR?

**Comment by ThreeMonth03** [comment]
Yes, I will. I will make a pr after [issue382](https://github.com/solvcon/modmesh/issues/382)

**Comment by yungyuc** [comment]
@ThreeMonth03 Do we have any work left for this issue?

**Comment by ThreeMonth03** [comment]
> @ThreeMonth03 Do we have any work left for this issue?

I think I should expose more class members for serializer and implement the deserializer, but I'm writing the notes for the newcomers.

**Comment by yungyuc** [comment]
> > @ThreeMonth03 Do we have any work left for this issue?
> 
> I think I should expose more class members for serializer and implement the deserializer, but I'm writing the notes for the newcomers.

No problem.  I added the two tasks in the issue description.  This issue remains open to track them.

**Comment by yungyuc** [comment]
@ThreeMonth03 could you please help review if there is remaining work with this issue?

**Comment by tigercosmos** [comment]
@ThreeMonth03  Could you help check if we can close this one?

**Comment by tigercosmos** [comment]
As far as I know, this issue should be already addressed. Closing now.
